### PR TITLE
FSx in isolated subnet

### PR DESCRIPTION
**Pull request by rexcsn**
* Do not query FSx API for MountName and DNSName
* For existing filesystems, MountName and DNSName are retrieved by CLI and passed into cookbook using attributes
* For filesystem created with cluster, MountName is retrieved from CFN filesystem resource attribute. DNSName is generated in hardcoded format
* Fix unit tests
* Add integration test for FSx in private subnet

Signed-off-by: Rex <shuningc@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/config/mappings.py [hunk 1]
```diff
@@ -575,7 +577,19 @@
                 "default": "NONE",
                 "allowed_values": ALLOWED_VALUES["fsx_drive_cache_type"],
                 "update_policy": UpdatePolicy.UNSUPPORTED
-            })
+            }),
+            ("existing_mount_name", {
+                "type": FSxMountNameParam,
+                "default": "NONE",
+                "update_policy": UpdatePolicy.UNSUPPORTED,
```
Comment (demartinofra) [inline cli/src/pcluster/config/mappings.py:584]: probably for private parameters we want to have IGNORED as UpdatePolicy and rely on the UNSUPPORTED update for fsx_fs_id?otherwise I don't know if this causes problems such as reporting them in the update diffs

File: cloudformation/aws-parallelcluster.cfn.json [hunk 2]
```diff
@@ -336,9 +336,9 @@
       "Default": "1"
     },
     "FSXOptions": {
-      "Description": "Comma separated list of FSx related options, 17 parameters in total, [shared_dir,fsx_fs_id,storage_capacity,fsx_kms_key_id,imported_file_chunk_size,export_path,import_path,weekly_maintenance_start_time,deployment_type,per_unit_storage_throughput,daily_automatic_backup_start_time,automatic_backup_retention_days,copy_tags_to_backups,fsx_backup_id,auto_import_policy,storage_type,drive_cache_type]",
+      "Description": "Comma separated list of FSx related options, 19 parameters in total, [shared_dir,fsx_fs_id,storage_capacity,fsx_kms_key_id,imported_file_chunk_size,export_path,import_path,weekly_maintenance_start_time,deployment_type,per_unit_storage_throughput,daily_automatic_backup_start_time,automatic_backup_retention_days,copy_tags_to_backups,fsx_backup_id,auto_import_policy,storage_type,drive_cache_type,existing_mount_name,existing_dns_name]",
```
Comment (demartinofra) [inline cloudformation/aws-parallelcluster.cfn.json:339]: not part of this PR but I'm wondering how close we are to reach the param value length limit when several fsx settings are used

File: cloudformation/tests/test_cfn_template_rendering.py [hunk 3]
```diff
@@ -199,7 +199,7 @@ def test_hit_substack_rendering(tmp_path, test_config):
             "vol1,vol2,vol3,vol4,vol5",
             "gp2,io1,sc1,st1,gp2",  # No io1 metrics
             "raid,1,2,gp2,20,100,false,NONE",
-            "/fsx,NONE,1200,NONE,NONE,NONE,NONE,NONE,NONE,NONE,NONE,NONE,NONE,NONE",
+            "/fsx,NONE,1200,NONE,NONE,NONE,NONE,NONE,NONE,NONE,NONE,NONE,NONE,NONE,NONE,NONE,NONE,NONE",
```
Comment (demartinofra) [inline cloudformation/tests/test_cfn_template_rendering.py:202]: was this test even running?cause I can see you had to add NONE 4 times

**Comment by demartinofra** [comment]
let's add a note to the changelog about supporting FSX in subnets with no internet access

File: cloudformation/tests/test_cfn_template_rendering.py (hunk 3 above)
Comment (rexcsn) [reply cloudformation/tests/test_cfn_template_rendering.py]: Test is running, but logic only depends on the first shared_dir parameter, so other FSx params do not matter.
Added a comment in the code to explain this

**Review by demartinofra — APPROVED** [review]
